kernel/base/ipc/los_rwlock.c

```diff
@@ -0,0 +1,499 @@
+    LOS_IntRestore(uwIntSave);
+    return LOS_OK;
+ErrHandler:
+    OS_RETURN_ERROR_P2(uwErrLine, uwErrNo);
+}
+
+/*****************************************************************************
+ Function     : LOS_RWLockDelete
+ Description  : Delete a rwlock,
+ Input        : uwRWLockHandle------Mutex operation handle
+ Output       : None
+ Return       : LOS_OK on success ,or error code on failure
+ *****************************************************************************/
+LITE_OS_SEC_TEXT_INIT UINT32 LOS_RWLockDelete(UINT32 uwRWLockHandle)
+{
+    UINT32    uwIntSave;
+    RWLOCK_CB_S *pstRWLockDeleted;
+    UINT32   uwErrNo;
+    UINT32   uwErrLine;
+
+    if (uwRWLockHandle >= (UINT32)LOSCFG_BASE_IPC_RWLOCK_LIMIT)
+    {
+        OS_GOTO_ERR_HANDLER(LOS_ERRNO_RWLOCK_INVALID);
+    }
+
+    pstRWLockDeleted = GET_RWLOCK(uwRWLockHandle);
+    uwIntSave = LOS_IntLock();
+    if (OS_RWLOCK_UNUSED == pstRWLockDeleted->ucRWLockStat)
+    {
+        LOS_IntRestore(uwIntSave);
+        OS_GOTO_ERR_HANDLER(LOS_ERRNO_RWLOCK_INVALID);
+    }
+
+    if ((!LOS_ListEmpty(&pstRWLockDeleted->stRLockList)) \
+        || (!LOS_ListEmpty(&pstRWLockDeleted->stRLockList)) \
+        || pstRWLockDeleted->ucRWLockCount)
+    {
+        LOS_IntRestore(uwIntSave);
+        OS_GOTO_ERR_HANDLER(LOS_ERRNO_RWLOCK_PENDED);
+    }
+
+    LOS_ListAdd(&g_stUnusedRWLockList, &pstRWLockDeleted->stRLockList);
+    pstRWLockDeleted->ucRWLockStat  = OS_RWLOCK_UNUSED;
+    pstRWLockDeleted->pstWOwner     = NULL;
+    pstRWLockDeleted->ucRWLockCount = 0;
+    pstRWLockDeleted->ucRCount      = 0;
+
+    LOS_IntRestore(uwIntSave);
+
+    return LOS_OK;
+ErrHandler:
+    OS_RETURN_ERROR_P2(uwErrLine, uwErrNo);
+}
+
+/*****************************************************************************
+ Function     : LOS_RLock
+ Description  : Specify the rwlock r lock operation,
+ Input        : uwRWLockHandle ------ RWLock operation handleone,
+                uwTimeOut  ------- waiting time,
+ Output       : None
+ Return       : LOS_OK on success ,or error code on failure
+ *****************************************************************************/
+LITE_OS_SEC_TEXT UINT32 LOS_RLock(UINT32 uwRWLockHandle, UINT32 uwTimeout)
+{
+    UINT32     uwIntSave;
+    RWLOCK_CB_S  *pstRWLockPended;
+    UINT32     uwRetErr;
+    LOS_TASK_CB  *pstRunTsk;
+
+    if (uwRWLockHandle >= (UINT32)LOSCFG_BASE_IPC_RWLOCK_LIMIT)
+    {
+        OS_RETURN_ERROR(LOS_ERRNO_RWLOCK_INVALID);
+    }
+
+    pstRWLockPended = GET_RWLOCK(uwRWLockHandle);
+    uwIntSave = LOS_IntLock();
+    if (OS_RWLOCK_UNUSED == pstRWLockPended->ucRWLockStat)
+    {
+        LOS_IntRestore(uwIntSave);
+        OS_RETURN_ERROR(LOS_ERRNO_RWLOCK_INVALID);
+    }
+
+    if (OS_INT_ACTIVE)
+    {
+        LOS_IntRestore(uwIntSave);
+        return LOS_ERRNO_RWLOCK_PEND_INTERR;
+    }
+
+    pstRunTsk = (LOS_TASK_CB *)g_stLosTask.pstRunTask;
+    if (pstRWLockPended->ucRWLockCount == 0)
```

Comment:
This if and the next one can be combined as:

if (WriteCount == 0)
{
    ...
    return OK;
}